perception/sub8_batcave_ogrid/src/OGridGen.cpp

```diff
@@ -0,0 +1,133 @@
+#include "OGridGen.hpp"
+
+//TODO: Add service call to clear ogrid
+
+OGridGen::OGridGen() : nh(ros::this_node::getName()), classification(&nh)
+{
+  pubGrid = nh.advertise<nav_msgs::OccupancyGrid>("/ogrid", 10, true);
+  pubPointCloud = nh.advertise<sensor_msgs::PointCloud2>("/point_cloud", 1);
+  nh.param<float>("resolution", resolution, 0.2f);
+  nh.param<float>("ogrid_size", ogrid_size, 50.f);
+  nh.param<float>("pool_depth", pool_depth, 7.f);
+  timer = nh.createTimer(ros::Duration(0.3), std::bind(&OGridGen::publish_ogrid, this, std::placeholders::_1));
+  matOgrid = cv::Mat::zeros((int)(ogrid_size / resolution), (int)(ogrid_size / resolution), CV_8U);
+
+  get_bounds = nh.serviceClient<sub8_msgs::Bounds>("get_bounds");
+
+  sub = nh.subscribe("/blueview_driver/ranges", 1, &OGridGen::callback, this);
+
+  int pointCloudBufferSize;
+  nh.param<int>("buffer_size", pointCloudBufferSize, 5000);
+  pointCloudBuffer.resize(pointCloudBufferSize);
+}
+
+void OGridGen::publish_ogrid(const ros::TimerEvent &)
+{
+  // Draw ogrid
+  sub8_msgs::Bounds bound_hold;
+  if (get_bounds.call(bound_hold))
+  {
+    bounds = {};
```

Comment:
You know how many bounds there are, so you know the size this vector needs to be. Itd be best to preallocate that.